web/modules/custom/dpl_paragraphs/dpl_paragraphs.module

```diff
@@ -159,15 +159,21 @@ function dpl_paragraphs_preprocess_paragraph__recommendation(array &$variables):
       // or description, otherwise the URL is constructed in the React app.
       $route_parameters = ['wid' => $work_id];
       $query_parameters = [];
+      $material_url = NULL;
 
-      if (!empty($material_type)) {
-        $query_parameters['type'] = $material_type;
-      }
+      try {
+        if (!empty($material_type)) {
+          $query_parameters['type'] = $material_type;
+        }
 
-      $material_url_object = Url::fromRoute('dpl_react_apps.work', $route_parameters,
-        ['absolute' => TRUE, 'query' => $query_parameters]);
+        $material_url_object = Url::fromRoute('dpl_react_apps.work', $route_parameters,
+          ['absolute' => TRUE, 'query' => $query_parameters]);
 
-      $material_url = $material_url_object->toString();
+        $material_url = $material_url_object->toString();
+      }
+      catch (\Exception $e) {
+        \Drupal::logger('dpl_paragraphs')->error('Could not render Recommendation paragraph - most likely cause the @wid WorkID is in wrong format.', ['@wid' => $work_id]);
+      }
```

Comment:
Please reconsider this structure.

1. It seems as if the change wraps some part of the code (L165-168) in the try catch which is not related to the problem. If that is the case then please keep it out.
2. Please catch the relevant exception class (or parent class if we think a group of exceptions is relevant) instead of the generic `\Exception` class. This makes it possible to have other problems fly under the radar which should be found during development.
3. Please include the error message from the exception in the logged message. Otherwise it is even more difficult to know what has happened.
4. Consider what should happen if this error occurs. The current approach passes `NULL` as the material url. I wonder what will happen in this case. The error message assumes that the recommendation paragraph is not rendered at all but is that actually the case? It seems as we create the render array for the React app anyway. If it should not be rendered I would not have expected this.
5. This addreses the problem at runtime. Do we need to add some kind of validation to the form elements, update hooks to fix existing problems or the like?

Reply:
1) Fixed
2) Fixed
3) Fixed
4) I set the URL to NULL, and then I would assume the material theme / react taking care of what should happen.

This is what happens - the skeleton shows up, and then an error shows up afterwards:
<img width="649" alt="Screenshot 2024-06-09 at 16 04 18" src="https://github.com/danskernesdigitalebibliotek/dpl-cms/assets/12376583/5aa0a263-c2fc-4962-860c-ede6560d360a">
<img width="1000" alt="Screenshot 2024-06-09 at 16 04 26" src="https://github.com/danskernesdigitalebibliotek/dpl-cms/assets/12376583/6a559d75-f97f-4f7c-84ab-4e186be19d12">

Which, is honestly what I would expect to happen - especially when you consider 5:

5) Nope, cause in a moment, we'll have the new backend where this is not an issue at all. I see this update mostly as a quickfix